components/manager/internal/services/update-deadline-by-id.go

```
@@ -0,0 +1,137 @@
+	pkgHTTP "github.com/LerianStudio/reporter/pkg/net/http"
+
+	"github.com/LerianStudio/lib-commons/v3/commons"
+	libOpentelemetry "github.com/LerianStudio/lib-commons/v3/commons/opentelemetry"
+	"github.com/google/uuid"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.opentelemetry.io/otel/attribute"
+)
+
+// UpdateDeadlineByID updates an existing deadline with partial fields and returns the updated entity.
+func (uc *UseCase) UpdateDeadlineByID(ctx context.Context, id uuid.UUID, input *deadline.UpdateDeadlineInput) (*deadline.Deadline, error) {
```

golangci-lint (2.5.0)

[error] cyclomatic complexity 19 of func `(*UseCase).UpdateDeadlineByID` is high (> 18)

(gocyclo)